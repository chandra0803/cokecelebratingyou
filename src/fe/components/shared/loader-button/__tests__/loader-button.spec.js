import React from 'react';
import LoaderButton from '../loader-button';
import renderer from 'react-test-renderer';

// Matrix of props that affect component rendering
const mockOptions = [
    { busy: false, disabled: true,  done: false, fetching: false },
    { busy: false, disabled: false, done: false, fetching: false },
    { busy: true,  disabled: false, done: false, fetching: false },
    { busy: false, disabled: false, done: true, fetching: false  },
    { busy: false, disabled: true,  done: false, fetching: true },
    { busy: false, disabled: false, done: false, fetching: true },
    { busy: true,  disabled: false, done: false, fetching: true },
    { busy: false, disabled: false, done: true, fetching: true  },
];

test( 'Loader button renders different icons depending on props passed in', () => {
    const component = renderer.create(
        <div>
            { mockOptions.map( opt => (
                <div>
                    <LoaderButton
                        busy={ opt.busy }
                        onClick={ () => true }
                        handleClick={ ()=>true }
                        disabled={ opt.disabled }
                        fetching={ opt.fetching }
                        done={ opt.done }>
                        Log in
                    </LoaderButton>
                </div>
            ) ) }
        </div>
    );

    const tree = component.toJSON();
    expect( tree ).toMatchSnapshot();
} );
