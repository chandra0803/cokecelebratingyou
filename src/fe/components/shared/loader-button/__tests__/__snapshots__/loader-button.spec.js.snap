// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader button renders different icons depending on props passed in 1`] = `
<div>
  <div>
    <button
      className="gq-btn disabled"
      onClick={[Function]}
      tabIndex=""
      type="button"
    >
      Log in
    </button>
  </div>
  <div>
    <button
      className="gq-btn"
      onClick={[Function]}
      tabIndex=""
      type="button"
    >
      Log in
    </button>
  </div>
  <div>
    <button
      className="gq-btn"
      onClick={[Function]}
      tabIndex=""
      type="button"
    >
      Log in
    </button>
  </div>
  <div>
    <button
      className="gq-btn done"
      onClick={[Function]}
      tabIndex=""
      type="button"
    >
      <div
        className="loader-container"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
    </button>
  </div>
  <div>
    <button
      className="gq-btn fetching disabled"
      onClick={[Function]}
      tabIndex=""
      type="button"
    >
      <div
        className="loader-container spinning"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.8,1.5h5.4c0.4,0,0.8-0.3,0.8-0.8C24,0.3,23.7,0,23.2,0h-7.5C15.7,0,15,0,15,0.8v7.5C15,8.7,15.3,9,15.8,9 s0.8-0.3,0.8-0.8V2.5c3.5,1.7,6,5.3,6,9.5c0,5.8-4.7,10.5-10.5,10.5S1.5,17.8,1.5,12c0-5.3,3.9-9.7,9-10.4V0.1C4.6,0.8,0,5.9,0,12 c0,6.6,5.4,12,12,12s12-5.4,12-12C24,7.5,21.5,3.5,17.8,1.5z"
          />
        </svg>
      </div>
    </button>
  </div>
  <div>
    <button
      className="gq-btn fetching"
      onClick={[Function]}
      tabIndex=""
      type="button"
    >
      <div
        className="loader-container spinning"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.8,1.5h5.4c0.4,0,0.8-0.3,0.8-0.8C24,0.3,23.7,0,23.2,0h-7.5C15.7,0,15,0,15,0.8v7.5C15,8.7,15.3,9,15.8,9 s0.8-0.3,0.8-0.8V2.5c3.5,1.7,6,5.3,6,9.5c0,5.8-4.7,10.5-10.5,10.5S1.5,17.8,1.5,12c0-5.3,3.9-9.7,9-10.4V0.1C4.6,0.8,0,5.9,0,12 c0,6.6,5.4,12,12,12s12-5.4,12-12C24,7.5,21.5,3.5,17.8,1.5z"
          />
        </svg>
      </div>
    </button>
  </div>
  <div>
    <button
      className="gq-btn fetching"
      onClick={[Function]}
      tabIndex=""
      type="button"
    >
      <div
        className="loader-container spinning"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.8,1.5h5.4c0.4,0,0.8-0.3,0.8-0.8C24,0.3,23.7,0,23.2,0h-7.5C15.7,0,15,0,15,0.8v7.5C15,8.7,15.3,9,15.8,9 s0.8-0.3,0.8-0.8V2.5c3.5,1.7,6,5.3,6,9.5c0,5.8-4.7,10.5-10.5,10.5S1.5,17.8,1.5,12c0-5.3,3.9-9.7,9-10.4V0.1C4.6,0.8,0,5.9,0,12 c0,6.6,5.4,12,12,12s12-5.4,12-12C24,7.5,21.5,3.5,17.8,1.5z"
          />
        </svg>
      </div>
    </button>
  </div>
  <div>
    <button
      className="gq-btn done"
      onClick={[Function]}
      tabIndex=""
      type="button"
    >
      <div
        className="loader-container"
      >
        <svg
          className="icon"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
          />
          <path
            d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;
